Image gallery

jleiandy opened the pull request:
Image scroll is working but could use some polish, will return to the expand and zoom feature after finishing product info component

jleiandy committed:
Added margin padding to main APP container,
refactored product image to be displayed in background,
added temporary template divs for future product info.

jleiandy committed:
Refactored styleindex, imageindex, and currentstyle to be held
in ProductOverview.
Added thumbnail scroller

jleiandy committed:
Merge branch 'main' into ImageGallery
Fixed some lint,
Refactored css files to scss files

jleiandy committed:
Implemented MVP of thumbnail and image scroll view
Edited test due to changing component design.

jleiandy committed:
Added mockdata to gitignore

jleiandy committed:
Delete MockData directory

jleiandy committed:
Merge branch 'main' into ImageGallery

jleiandy committed:
Editted tests to search for ProductOverview container
will set up mock axios request in future.

jleiandy committed:
Merge branch 'ImageGallery' of https://github.com/HRSF137-FEC-Enoki/catwalk into ImageGallery

cschucode commented on client/components/ProductOverview/ImageGallery.jsx:
```diff
@@ -1,26 +1,39 @@
+  }
+  const currentPhoto = currentStyle.photos[imageIndex].url;
 
   const handleLeftArrow = () => {
```
Couldn't we have all of the logic live in one handler with a flag of either left or right?

jleiandy replied:
refactored

cschucode commented on client/components/ProductOverview/ImageGallery.jsx:
```diff
@@ -1,26 +1,39 @@
+    <div className="imageGalleryContainer">
+      <div className="mainImageContainer" style={{ backgroundImage: `url(${currentPhoto})` }} />
+      <ThumbnailView photos={photos} handleClick={handleThumbNailClick} className="thumbnailComponent" currentIndex={imageIndex} />
+      <AiOutlineArrowLeft className="leftArrow" onClick={handleLeftArrow} size={35} />
```
Can we create a variable for the size and store it at the top of the component? Something like `const ARROW_SIZE = 35;` That way if anything changes, you'd only have to change it in one place.

jleiandy replied:
refactored

cschucode commented on client/components/ProductOverview/ProductOverview.jsx:
```diff
@@ -1,17 +1,38 @@
 import ImageGallery from './ImageGallery';
 
-import '../../css/ProductOverviewCSS.css';
+import '../../css/ProductOverviewCSS.scss';
```
Just a naming thing, but `ProductOverviewCSS` seems a little redundant considering it's in a `css` directory and it's got a `.scss`  file extension. Also, it's best practice to name your css files after the component that imports them. Here's an example from the docs https://reactjs.org/docs/faq-structure.html

jleiandy replied:
i took out the CSS in the file name. I will start creating separate scss files from now on for each of my components.

cschucode commented on client/components/ProductOverview/ProductOverview.jsx:
```diff
@@ -1,17 +1,38 @@
+  const [currentStyle, setCurrentStyle] = useState(null);
+
+  useEffect(() => {
+    axios.get('/products/48436/styles')
```
Is there a way to pass the product_id in here without hard coding?

jleiandy replied:
Refactored, I was initially using that specific product id because it had more than 7 thumbnails that I could use to test the functionality of my thumbnail scroller

cschucode commented on client/components/ProductOverview/ThumbnailView.jsx:
```diff
@@ -0,0 +1,76 @@
+    }
+  }, [currentIndex]);
+
+  const handleArrowUp = () => {
```
Same with arrowLeft and arrowRight. Is there a way to have one handler that changes behavior based on a flag rather than additional functions?

jleiandy replied:
I think in this case the code is more readable with separate functions.

cschucode commented on client/components/ProductOverview/ThumbnailView.jsx:
```diff
@@ -0,0 +1,76 @@
+
+  return (
+    <div className="arrowsThumbnailContainer">
+      {currentIndex !== 0 ? <AiOutlineArrowUp className="upArrow" size={45} onClick={handleArrowUp} /> : null}
```
I would make the size prop more extensible by using a variable instead of a magic number out of thin air. It would definitely make it more readable and you'd only have to change it in once place in the future if anything changes.

https://betterprogramming.pub/stop-using-magic-numbers-and-variables-in-your-code-4e86f008b84c

jleiandy replied:
added arrowsize variable

cschucode commented on client/css/App.scss:
```diff
@@ -1,9 +1,15 @@
 #app {
   font-family: Arial, Helvetica, sans-serif;
+  border: solid 2px black;
+  margin: 0 7%;
```
Percentages are hard to work with. Let's figure out a `px` or `em` width for consistency.

jleiandy replied:
I am not that familiar with the benefits of each size unit. I thought that using percentages would account for the size of the screen automatically. I think we should have a discussion as a group to discuss a standard.

cschucode commented on client/css/ProductOverviewCSS.scss:
```diff
@@ -0,0 +1,94 @@
+
+.productDescription {
+  border: 2px solid black;
+  margin: 1% 0;
```
I cringe a little every time I see these percentages. I think a set value would be better and then a media query for different screen sizes would be best.

jleiandy replied:
I think we should discuss sizing as a group. Not the most important right now for MVP. Media queries are a good idea, not sure on the best practices of implementing them though.

cschucode commented on client/css/ProductOverviewCSS.scss:
```diff
@@ -0,0 +1,94 @@
+  border: 3px solid blue;
+}
+
+.upArrow {
```
can we combine these selectors since they're doing the same thing?

jleiandy replied:
Combined

cschucode commented on jest.config.js:
```diff
@@ -86,7 +86,7 @@ module.exports = {
   // A map from regular expressions to module names or to arrays of module names that
   // allow to stub out resources with a single module
-  moduleNameMapper: { '\\.(css|less)$': '<rootDir>/styleMock.js' },
+  moduleNameMapper: { '\\.(scss|less)$': '<rootDir>/styleMock.js' },
```
Nice 👍

cschucode commented on server/index.js:
```diff
@@ -49,6 +49,18 @@ app.get('/api/products/:product_id', (req, res) => {
     });
 });
 
+app.get('/products/:product_id/styles', (req, res) => {
```
I think the only difference in our two routes is that we're now using `/api/proucts/....` for our routes that hit the API.

jleiandy replied:
added /api/ to front of endpoint

jleiandy committed:
Combined arrow handlers into one function
Reorganized css files

jleiandy committed:
Uses live axios call to get productId

jleiandy committed:
Added arrowsize variable for up and down arrows
combined styling for up and down arrows
added /api/ to beginnign of endpoint calls.

cschucode commented on .gitignore:
```diff
@@ -6,3 +6,7 @@ public/bundle.js
 # jest coverage
 coverage
+#MockData used for ProductOverview component
+MockData
```
Do we still need this if the file was deleted?

jleiandy replied:
It's not deleted on my local repo, I am thinking I might still need mockdata for future tests.

cschucode replied:
It's not hurting anything, so we can leave it.

cschucode reviewed (approved):
Looks okay. Nice job on the refactor. Some points like % vs px are a matter of preference. We can definitely discuss further.
